File: №2_4.cpp
#include <iostream>
using namespace std;

int main() {
    int N, V;
    cout << "Введите N и V: ";
    cin >> N >> V;
    
    int fuel = V;
    int stops = 0;
    
    cout << "Введите расстояния R: ";
    for (int i = 0; i < N - 1; i++) {
        int R;
        cin >> R;
        
        if (R > V) {
            cout << 0 << endl;
            return 0;
        }
        
        if (fuel < R) {
            stops++;
            fuel = V;
        }
        
        fuel -= R;
    }
    
    cout << stops << endl;
    return 0;
}

File: №2_4.go
package main

import "fmt"

func main() {
    var N, V int
    fmt.Print("Введите N и V: ")
    fmt.Scan(&N, &V)
    
    fuel := V
    stops := 0
    
    fmt.Print("Введите расстояния R: ")
    for i := 0; i < N-1; i++ {
        var R int
        fmt.Scan(&R)
        
        if R > V {
            fmt.Println(0)
            return
        }
        
        if fuel < R {
            stops++
            fuel = V
        }
        
        fuel -= R
    }
    
    fmt.Println(stops)
}